core/unit_test/TestRange.hpp

```diff
@@ -327,6 +327,71 @@ TEST(TEST_CATEGORY, range_reduce) {
   }
 }
 
+template <typename ExecSpace, typename StaticBatchSize>
+struct TestStaticBatchSize {
+  using value_type = int;  ///< alias required for the parallel_reduce
+  using view_type  = Kokkos::View<value_type *, ExecSpace>;
+
+  view_type m_flags;
+  view_type result_view;
+
+  struct AtomicAddTag {};
+  struct VerifyAtomicAddTag {};
+
+  int N;
+
+  TestStaticBatchSize(const size_t N_)
+      : m_flags(Kokkos::view_alloc(Kokkos::WithoutInitializing, "flags"), N_),
+        result_view(Kokkos::view_alloc(Kokkos::WithoutInitializing, "results"),
+                    N_),
+        N(N_) {}
+
+  void test_batch_size() {
+    typename view_type::HostMirror host_flags =
+        Kokkos::create_mirror_view(m_flags);
+
+    // Initialize the flags
+    Kokkos::parallel_for(Kokkos::RangePolicy<ExecSpace>(0, N), *this);
```

Comment:
This would make your test easier to read:
```suggestion
    Kokkos::deep_copy(m_flags, 0);
```
(and get rid of the untagged call operator below)